frontend/src/metabase/public/components/widgets/EmbedModal.tsx

```diff
@@ -35,14 +41,29 @@ export const EmbedModal = ({
   onClose: () => void;
   children: ({
     embedType,
-    setEmbedType,
+    goToNextStep,
+    goToPreviousStep,
   }: {
     embedType: EmbedModalStep;
-    setEmbedType: (type: EmbedModalStep) => void;
+    goToNextStep: () => void;
+    goToPreviousStep: () => void;
   }) => JSX.Element;
 } & WindowModalProps) => {
   const [embedType, setEmbedType] = useState<EmbedModalStep>(null);
-  const applicationName = useSelector(getApplicationName);
+
+  const goToNextStep = () => {
```

Comment:
This function will be expanded to check if the user has already accepted the legalese in the future. Right now it's just cycling through the states